### PR TITLE
custom uni v3 router for the cash liquidations data

### PR DESCRIPTION
**Pull request by vminkov**
## Description
custom uni v3 router for the cash liquidations data

### CODE REVIEW

File: contracts/liquidators/registry/LiquidatorsRegistryExtension.sol [hunk 1]
```diff
@@ -583,6 +601,18 @@ contract LiquidatorsRegistryExtension is LiquidatorsRegistryStorage, DiamondExte
...
   }
 
+
+  function uniswapV3LiquidatorData(IERC20Upgradeable inputToken, IERC20Upgradeable outputToken)
+    internal
+    view
+    returns (bytes memory strategyData)
+  {
+    // TODO
+    uint24 fee;
```
Comment (vminkov) [inline contracts/liquidators/registry/LiquidatorsRegistryExtension.sol]: @carlomazzaferro do you have any idea about the fee tier here?

File: contracts/liquidators/registry/LiquidatorsRegistryExtension.sol [hunk 2]
```diff
@@ -511,6 +513,22 @@ contract LiquidatorsRegistryExtension is LiquidatorsRegistryStorage, DiamondExte
     return tokens[0];
   }
 
+  function getUniswapV3Router(IERC20Upgradeable inputToken, IERC20Upgradeable outputToken) internal view returns (address) {
+    address CASH = 0x5D066D022EDE10eFa2717eD3D79f22F949F8C175;
+    address USDC = 0x2791Bca1f2de4661ED88A30C99A7a9449Aa84174;
+
+    if (
+      (address(inputToken) == USDC && address(outputToken) == CASH)
```
Comment (carlomazzaferro) [inline contracts/liquidators/registry/LiquidatorsRegistryExtension.sol]: We will need a bunch of these statements. How can we optimise this? We'll have at least 5-6 LP tokens with such configuration

Comment (vminkov) [reply contracts/liquidators/registry/LiquidatorsRegistryExtension.sol]: Probably better with a storage mapping?

Comment (carlomazzaferro) [reply contracts/liquidators/registry/LiquidatorsRegistryExtension.sol]: let's discuss in discord

File: contracts/liquidators/registry/LiquidatorsRegistryExtension.sol [hunk 3]
```diff
@@ -458,6 +472,8 @@ contract LiquidatorsRegistryExtension is LiquidatorsRegistryStorage, DiamondExte
       strategyData = solidlyLpTokenLiquidatorData(inputToken, outputToken);
     } else if (isStrategy(strategy, "UniswapV2LiquidatorFunder")) {
       strategyData = uniswapV2LiquidatorData(inputToken, outputToken);
+    } else if (isStrategy(strategy, "UniswapV3Liquidator")) {
+      strategyData = uniswapV3LiquidatorData(inputToken, outputToken);
     } else if (isStrategy(strategy, "AlgebraSwapLiquidator") || isStrategy(strategy, "GammaLpTokenLiquidator")) {
```
Comment (vminkov) [inline contracts/liquidators/registry/LiquidatorsRegistryExtension.sol]: TODO: separate

`    } else if (isStrategy(strategy, "AlgebraSwapLiquidator") || isStrategy(strategy, "GammaLpTokenLiquidator")) {
`

**Review by carlomazzaferro — APPROVED** [review]

**Pull request merged**